plugins/inputs/nvidia_smi/nvidia_smi.go

```diff
@@ -31,20 +32,34 @@ func (smi *NvidiaSMI) Description() string {
 // SampleConfig returns the sample configuration for the NvidiaSMI plugin
 func (smi *NvidiaSMI) SampleConfig() string {
 	return `
-  ## Optional: path to nvidia-smi binary, defaults to $PATH via exec.LookPath
-  # bin_path = "/usr/bin/nvidia-smi"
+  ## Optional: path to nvidia-smi binary, if not specified or an empty string is specified, we will search for it on exec.LookPath
+  ## We will use "os.Stat" to check whether the nvidia-smi binary actually exists at the location pointed to by the bin_path, 
+  ## if not, an error will be returned as soon as possible (in Init())
+  # bin_path = ""
 
-  ## Optional: timeout for GPU polling
+  ## Optional: timeout for GPU polling, defaults "5s"
   # timeout = "5s"
 `
 }
 
-// Gather implements the telegraf interface
-func (smi *NvidiaSMI) Gather(acc telegraf.Accumulator) error {
+func (smi *NvidiaSMI) Init() error {
+	if smi.BinPath == "" {
+		var err error
+		smi.BinPath, err = exec.LookPath("nvidia-smi")
+		if err != nil {
+			return errors.New("nvidia-smi not found: verify that nvidia-smi is installed and that nvidia-smi is in your PATH")
+		}
+	}
+	// fail-fast
 	if _, err := os.Stat(smi.BinPath); os.IsNotExist(err) {
-		return fmt.Errorf("nvidia-smi binary not at path %s, cannot gather GPU data", smi.BinPath)
+		return fmt.Errorf("nvidia-smi binary not at path %s", smi.BinPath)
 	}
```

Comment:
Can we please make this backward compatible using
```golang
	// fail-fast
	if _, err := os.Stat(smi.BinPath); os.IsNotExist(err) {
		path, patherr := exec.LookPath("nvidia-smi")
		if err != nil {
			return errors.New("nvidia-smi not found in %q and not in PATH; please make sure nvidia-smi is installed and/or is in PATH", smi.BinPath)
		}
		smi.BinPath = path
	}
```
and keeping the default for the binary at `/usr/bin/nvidia-smi`?